src/cheri_insts.sail

```diff
@@ -882,99 +706,52 @@ union clause ast = CBuildCap : (regidx, regidx, regidx)
  *   operation.
  */
 function clause execute (CBuildCap(cd, cs1, cs2)) = {
-  let (cs1_val, cs1_idx) = if unsigned(cs1) == 0 then (DDC, DDC_IDX)
-                                                 else (C(cs1), 0b0 @ cs1);
-  let cs2_val = C(cs2);
-  let (cs1_base, cs1_top) = getCapBounds(cs1_val);
-  let (cs2_base, cs2_top) = getCapBounds(cs2_val);
-  let cs1_perms = getCapPerms(cs1_val);
-  let cs2_perms = getCapPerms(cs2_val);
-  let cs2_flags = getCapFlags(cs2_val);
-  if not (cs1_val.tag) then {
-    handle_cheri_cap_exception(CapEx_TagViolation, cs1_idx);
-    RETIRE_FAIL
-  } else if isCapSealed(cs1_val) then {
-    handle_cheri_cap_exception(CapEx_SealViolation, cs1_idx);
-    RETIRE_FAIL
-  } else if cs2_base < cs1_base then {
-    handle_cheri_cap_exception(CapEx_LengthViolation, cs1_idx);
-    RETIRE_FAIL
-  } else if cs2_top > cs1_top then {
-    handle_cheri_cap_exception(CapEx_LengthViolation, cs1_idx);
-    RETIRE_FAIL
-  } else if cs2_base > cs2_top then { /* check for length < 0 - possible because cs2 might be untagged */
-    handle_cheri_reg_exception(CapEx_LengthViolation, cs2);
-    RETIRE_FAIL
-  } else if (cs2_perms & cs1_perms) != cs2_perms then {
-    handle_cheri_cap_exception(CapEx_UserDefViolation, cs1_idx);
-    RETIRE_FAIL
-  } else {
-    let (exact, cd1) = setCapBounds(cs1_val, to_bits(cap_addr_width, cs2_base), to_bits(cap_len_width, cs2_top));
-    let (_, cd2) = setCapOffset(cd1, getCapOffsetBits(cs2_val)); // Ignore representability check, since Fast Rep Check not relevant
-    let cd3 = setCapPerms(cd2, cs2_perms);
-    let cd4 = setCapFlags(cd3, cs2_flags);
-    if cd4 != {unsealCap(cs2_val) with tag=true} then
-    {
-      handle_cheri_reg_exception(CapEx_LengthViolation, cs2);
-      RETIRE_FAIL
-    } else {
-      assert(exact, "CBuildCap: setCapBounds was not exact"); /* base and top came from cs2 originally so will be exact */
-      let cd5 = if signed(cs2_val.otype) == otype_sentry then sealCap(cd4, to_bits(cap_otype_width, otype_sentry)) else cd4;
-      C(cd) = cd5;
-      RETIRE_SUCCESS
-    }
-  }
+  let authority_val = clearTagIfSealed(if unsigned(cs1) == 0 then DDC else C(cs1));
```

Comment:
This is a bit of a wall of code. Some blank lines to separate reading registers from checks from the actual operations might go a long way to help that.

Reply:
Hopefully sorted, and roughly in keeping with the new whitespace in other non-trivial instructions